alica_engine/src/engine/AlicaTimer.cpp

```diff
@@ -0,0 +1,68 @@
+#include "engine/AlicaTimer.h"
+
+namespace alica
+{
+
+class AlicaSystemTimer::AlicaSystemTimerImpl
+{
+public:
+    using TimerCb = std::function<void()>;
+
+    AlicaSystemTimerImpl(TimerCb&& userCb, alica::AlicaTime period)
+            : _userCb(std::move(userCb))
+            , _period(period.inMilliseconds())
+            , _isActive(false)
+    {
+    }
+
+    ~AlicaSystemTimerImpl()
+    {
+        if (_isActive) {
+            stop();
+        }
+    }
+
+    void start()
+    {
+        if (_isActive) {
+            stop();
+        }
+        _isActive = true;
+        _thread = std::thread([this]() {
+            while (_isActive) {
+                _userCb();
+                int64_t sleep_duration = _period;
+                while ((sleep_duration > 0) && _isActive) {
+                    std::this_thread::sleep_for(std::chrono::milliseconds(std::min(int64_t(500), _period)));
+                    sleep_duration -= std::min(int64_t(500), _period);
```

Comment:
1. `std::min(int64_t(500), _period)` is constant
2. strongly speaking we should check how long we slept and subtract it

Reply:
you're correct, ok

Reply:
this is still wrong, I realized the other while and I'm correcting it.

Reply:
corrected now, take a look